src/types/web-resource.ts

```diff
@@ -0,0 +1,155 @@
+import * as TypeUtils from '../type-utils';
+import { ContentOptions, getStorageAdapter } from '../storage-adapters/index';
+
+/**
+ * This is how WebResources are stored on the DB
+ */
+export type WebResourceRef = {
+	filename: string;
+	href: string;
+	contentType?: string;
+	contentDisposition?: string;
+	size?: number;
```

Comment:
In combination with the SBVR these three properties cannot be undefined.

Reply:
:+1:

Reply:
Fixed with:
```
Fact type:  WebResource has Content Type
	Necessity: Each WebResource has at most one Content Type.
Fact type:  WebResource has Content Disposition
	Necessity: Each WebResource has at most one Content Disposition.
Fact type:  WebResource has Size
	Necessity: Each WebResource has at most one Size.
```